fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -3529,6 +3551,16 @@ struct RestoreRangeTaskFunc : RestoreFileTaskFuncBase {
 		return returnStr;
 	}
 
+	ACTOR static Future<Void> _validTenantAccess(KeyRef key, Reference<TenantEntryCache<Void>> tenantCache) {
+		if (isSystemKey(key)) {
+			return Void();
+		}
+		state int64_t tenantId = TenantAPI::extractTenantIdFromKeyRef(key);
+		Optional<TenantEntryCachePayload<Void>> payload = wait(tenantCache->getById(tenantId));
+		ASSERT(payload.present());
```

Comment:
do we want to assert here?

Reply:
This check essentially ensures that each key we're writing to the database (in simulation) during restore phase belongs to a valid tenant in required tenant mode. This should always pass since we filter out mutations from deleted tenants but it is more so a sanity check. I can probably add a trace here as well to give more information

Reply:
Sounds good. thanks